### DISCUSSION TITLE
Quadratic Programming with box constraints

### DISCUSSION THREAD

**Discussion by Matth-Sch** [Simple-Robotics/proxsuite#259] [Q&A]
Hello, 

I wanted to use an augmented lagrangian optimization solver which works with box constraints so I decided to try proxsuite. 
I see in the "example" part of the github that box constraints should be initialized as follows : 
```
qp = proxsuite.proxqp.dense.QP(n, n_eq, n_in, True)
print(f"the qp is box constrained: {qp.is_box_constrained()}")
H, g, A, b, C, u, l = generate_mixed_qp(n)
l_box = -np.ones(n) * 1.0e10
u_box = np.ones(n) * 1.0e10
qp.init(H, g, A, b, C, l, u, l_box=l_box, u_box=u_box)
```

However, when I run this code I get an __init__ error telling me that a boolean is incompatible with the function arguments and that it should only take integers for n, n_eq and n_in. Can somebody explain me what I am doing wrong and if maybe I forgot to add something in my code ? 

Additional question: In my problem, I only have box constraints, so there sould be no problem in having n_eq and n_in = 0 in my case, right ?

Thanks a lot for your work and your time,

**Comment by fabinsch** [comment]
Hi @Matth-Sch ,
thanks for you interest in proxsuite. You are referring to the example [init_dense_qp_with_box.py](https://github.com/Simple-Robotics/proxsuite/blob/main/examples/python/init_dense_qp_with_box.py). The API shown in this example works well for me with `proxsuite v0.4.1`. You can get this latest version easily by `conda install proxsuite -c conda-forge` or by `pip install proxsuite`. How did you install proxsuite / which version are you using ? 

There is no problem in having only box constraints, see this extract here from an ipython session:
```
In [1]: import proxsuite
   ...: import numpy as np
   ...: import scipy.sparse as spa
   ...: 
   ...: 
   ...: def generate_mixed_qp(n, seed=1):
   ...:     # A function for generating random convex qps
   ...: 
   ...:     np.random.seed(seed)
   ...:     n_eq = int(n / 4)
   ...:     n_in = int(n / 4)
   ...:     m = n_eq + n_in
   ...: 
   ...:     P = spa.random(
   ...:         n, n, density=0.075, data_rvs=np.random.randn, format="csc"
   ...:     ).toarray()
   ...:     P = (P + P.T) / 2.0
   ...: 
   ...:     s = max(np.absolute(np.linalg.eigvals(P)))
   ...:     P += (abs(s) + 1e-02) * spa.eye(n)
   ...:     P = spa.coo_matrix(P)
   ...:     q = np.random.randn(n)
   ...:     A = spa.random(m, n, density=0.15, data_rvs=np.random.randn, format="csc").toarray()
   ...:     v = np.random.randn(n)  # Fictitious solution
   ...:     delta = np.random.rand(m)  # To get inequality
   ...:     u = A @ v
   ...:     l = -1.0e20 * np.ones(m)
   ...: 
   ...:     return P.toarray(), q, A[:n_eq, :], u[:n_eq], A[n_in:, :], u[n_in:], l[n_in:]
   ...: 
   ...: 
   ...: # load a qp object using qp problem dimensions
   ...: n = 10
   ...: n_eq = 0
   ...: n_in = 0
   ...: qp = proxsuite.proxqp.dense.QP(n, n_eq, n_in, True)
   ...: # you can check that there are constraints with method is_box_constrained
   ...: print(f"the qp is box constrained: {qp.is_box_constrained()}")
   ...: # generate a random QP
   ...: H, g, A, b, C, u, l = generate_mixed_qp(n)
   ...: l_box = -np.ones(n) * 1.0e10
   ...: u_box = np.ones(n) * 1.0e10
   ...: # initialize the model of the problem to solve
   ...: qp.init(H, g, None, None, None, None, None, l_box, u_box)
   ...: qp.solve()
the qp is box constrained: True

In [2]: print(proxsuite.__version__)
0.4.1
```

**Reply by Matth-Sch** [reply]
Hello, thanks for your response. 
I installed it using pip install proxsuite and checked that I have the 0.4.1 version. I created a new environment to see if it worked and now I have another problem when I import proxsuite with a module called "instructionset" in __init__.py : 
```
Traceback (most recent call last):
  File "c:\Users\username\Documents\\script.py", line 4, in <module>
    import proxsuite
  File "C:\Users\username\.conda\envs\Env_project\lib\site-packages\cmeel.prefix\lib\python3.9\site-packages\proxsuite\__init__.py", line 9, in <module>
    from . import instructionset
ImportError: DLL load failed while importing instructionset: Le module spécifié est introuvable.
```
Have you had reports of people having the same problem ? I keep installing and reinstalling proxsuite but nothing changes. And I do not know anything about DLL dependencies so I do not want to do mistakes. 
Thanks again

**Reply by fabinsch** [reply]
Hello, I cannot reproduce your problem. From your output, it looks like you are on a Windows machine. On a Windows machine I did the following
1)
```
conda create -n test-conda
conda activate test-conda
conda install proxsuite -c conda-forge
python -c "import proxsuite"
```

2)
```
conda create -n test-pip
conda activate test-pip
pip install proxsuite
python -c "import proxsuite"
```
both are working fine. Could you try to follow these steps?
